compiler/rustc_hir_typeck/src/gather_locals.rs

```diff
@@ -162,7 +164,9 @@ impl<'a, 'tcx> Visitor<'tcx> for GatherLocalsVisitor<'a, 'tcx> {
                     );
                 }
             } else {
-                if !self.fcx.tcx.features().unsized_locals {
+                if !(self.fcx.tcx.features().unsized_locals
```

Comment:
Similarly, this causes us to accept unsized locals if `repr_scalable` is enabled. This doesn't seem correct.